net/net.go

```diff
@@ -279,7 +284,7 @@ func (n *net) Close() (err error) {
 			log.Errorf("error closing connection: %v", err)
 		}
 	}
-	n.rpc.GracefulStop()
+	tu.StopGRPCServer(n.rpc)
```

Comment:
I added a util that kills the server as some time instead of just waiting for it to gracefully close, which can never happen. The tests were hanging once in awhile, I think it was due to this.